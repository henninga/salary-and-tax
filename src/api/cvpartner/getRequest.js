import { config } from 'dotenv';
import rp from 'request-promise';

config();

export const BASE_URL = `https://technologic.cvpartner.com`;

const getRequest = async url => await rp({
    url: BASE_URL + url, json: true, headers: { 'Authorization': `Token token="${process.env.CV_PARTNER_API_KEY}"` },
});

export default getRequest;
