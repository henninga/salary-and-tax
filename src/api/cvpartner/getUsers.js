import debug from 'debug';
import getRequest from './getRequest';

const logger = debug('technologic.as:getUsers');

const getUsers = () => {
    return getRequest(`/api/v1/users?offset=0`)
        .then(users => users.map(({ id, default_cv_id, name }) => ({ userId: id, cvId: default_cv_id, name })))
        .catch(err => logger('failed to get users', err));
};

export default getUsers;