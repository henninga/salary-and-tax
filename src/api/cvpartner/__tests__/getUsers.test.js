import getRequest from '../getRequest';
import getUsers from '../getUsers';

const mockFirstUser = { name: 'CV Partner', id: 'userId', default_cv_id: 'cvId' };
const mockSecondUser = { name: 'Partner CV', id: 'idUser', default_cv_id: 'idCv' };

getRequest.mockImplementation(() => Promise.resolve([
    mockFirstUser,
    mockSecondUser,
]));

jest.mock('../getRequest');

describe('getUsers', () => {
    it('should get users', async () => {
        await expect(getUsers()).resolves.toMatchSnapshot();
    });
    it('should use correct url', async () => {
        await getUsers();
        expect(getRequest).toHaveBeenCalledWith('/api/v1/users?offset=0');
    });
    it('should map default_cv_id to cv_id and id to userId', async () => {
        await expect(getUsers()).resolves.toEqual(expect.arrayContaining([
            expect.objectContaining({ cvId: mockFirstUser.default_cv_id, userId: 'userId' }),
            expect.objectContaining({ cvId: mockSecondUser.default_cv_id, userId: 'idUser' }),
        ]));
    });
});
