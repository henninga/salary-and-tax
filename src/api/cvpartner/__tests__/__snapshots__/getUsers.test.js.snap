// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUsers should get users 1`] = `
Array [
  Object {
    "cvId": "cvId",
    "name": "CV Partner",
    "userId": "userId",
  },
  Object {
    "cvId": "idCv",
    "name": "Partner CV",
    "userId": "idUser",
  },
]
`;
